label_studio/tasks/models.py

```diff
@@ -876,29 +876,28 @@ def bulk_update_stats_project_tasks(tasks, project=None):
     # get project if it's not in params
     if project is None:
         project = tasks[0].project
-    with transaction.atomic():
-        use_overlap = project._can_use_overlap()
-        maximum_annotations = project.maximum_annotations
-        # update filters if we can use overlap
-        if use_overlap:
-            # finished tasks
-            finished_tasks = tasks.filter(Q(total_annotations__gte=maximum_annotations) |
-                                          Q(total_annotations__gte=1, overlap=1))
-            ids = finished_tasks.values_list('id', flat=True)
-            tasks.filter(id__in=ids).update(is_labeled=True)
-            # unfinished tasks
-            tasks.exclude(id__in=ids).update(is_labeled=False)
-        else:
-            # update objects without saving if we can't use overlap
-            for task in tasks:
-                update_task_stats(task, save=False)
-            try:
-                # start update query batches
-                bulk_update(tasks, update_fields=['is_labeled'], batch_size=settings.BATCH_SIZE)
-            except OperationalError as exp:
-                logger.error("Operational error while updating tasks: {exc}", exc_info=True)
-                # try to update query batches one more time
-                start_job_async_or_sync(bulk_update, tasks, in_seconds=settings.BATCH_JOB_RETRY_TIMEOUT, update_fields=['is_labeled'], batch_size=settings.BATCH_SIZE)
+
+    use_overlap = project._can_use_overlap()
+    maximum_annotations = project.maximum_annotations
+    # update filters if we can use overlap
+    if use_overlap:
+        # finished tasks
+        finished_tasks = tasks.filter(Q(total_annotations__gte=maximum_annotations) |
+                                        Q(total_annotations__gte=1, overlap=1))
+        finished_tasks_ids = finished_tasks.values_list('id', flat=True)
+        tasks.update(is_labeled=Q(id__in=finished_tasks_ids))
```

Comment:
Query that this emits (I wasn't sure that a Q object could be used this way, but it seems that it can):

```
DEBUG    django.db.backends:utils.py:123 (0.000) UPDATE "task" SET "is_labeled" = ("task"."id" IN (SELECT U0."id" FROM "task" U0 WHERE (U0."project_id" = 1 AND (U0."total_annotations" >= 1 OR (U0."overlap" = 1 AND U0."total_annotations" >= 1))))) WHERE "task"."project_id" = 1; args=(1, 1, 1, 1, 1)
```